dts/riscv/niosv/niosv-g.dtsi

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2023, Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+
+/ {
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "intel,niosv";
+			reg = <0>;
+			clock-frequency = <50000000>;
+
+			/* Platform interrupts IRQs index start from 16 */
```

Comment:
Uhm, are you in this exact situation then? https://github.com/zephyrproject-rtos/zephyr/pull/59035

Reply:
@carlocaione , I have to add the offset manual in device tree. Nice patch from you to enable this, hope it go in main line soon.

Reply:
@sweeaun yeah, that is not going to work with this patch then. In my use-case the offset-ted lines are not practically usable anymore. What is your actual configuration? You have CLINT + CLIC?

Reply:
Yes. we do hart local interrupt and Platform-Level Interrupt Controller (PLIC).

Reply:
oh ok, you have CLINT + PLIC. In that case *I think* that you can use `CONFIG_2ND_LEVEL_INTERRUPTS` to get rid of the 16 lines offset.

Reply:
Hi @carlocaione , our core doesn't have multiple level and only 1 level for all interrupts. Just that the interrupt 16 to 31 is named as platform interrupt which is not another level of interrupt controller. Sorry for confusion.
https://www.intel.com/content/www/us/en/docs/programmable/683632/23-1/interrupt-controller-91908.html
bit 3 is for software interrupt
bit 7 is for machine timer
bit[31:16] for platform interrupt

Reply:
interesting. Out of curiosity: how the platform is managing `mcause`? For example `mcause` for software interrupt is carrying the id `3` and `19` for the third platform interrupt?